inference-engine/src/multi_device/utils/log.hpp

```diff
@@ -0,0 +1,214 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+#ifndef MULTIDEVICEPLUGIN_LOG_H
+#define MULTIDEVICEPLUGIN_LOG_H
+
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <string>
+#include <mutex>
+
+#include "singleton.hpp"
+#include "time_utils.hpp"
+#include "thread_utils.hpp"
+
+#ifdef COLOR_LOG
+#define COL(x) "\033[1;" #x ";40m"
+#define COL_END "\033[0m"
+#else
+#define COL(x) ""
+#define COL_END ""
+#endif
+
+#define RED COL(31)
+#define GREEN COL(32)
+#define YELLOW COL(33)
+#define BLUE COL(34)
+#define MAGENTA COL(35)
+#define CYAN COL(36)
+#define WHITE COL(0)
+#define DEFAULT_COLOR ""
+
+#ifdef ERROR
+#undef ERROR
+#endif
+
+namespace MultiDevicePlugin {
+inline int parseInteger(const char* str) {
+    std::string var(str ? str : "");
+    try {
+        return std::stoi(var);
+    } catch (...) {
+        return INT32_MAX;
+    }
+}
+
+inline std::string getFileName(const std::string& filePath) {
+    auto index = filePath.find_last_of("/\\");
+    if (std::string::npos == index) {
+        return filePath;
+    }
+    return filePath.substr(index + 1);
+}
+
+inline int getDebugLevel() {
+    return 0;
+    //if want to set log level with env Variable, use below code.
```

Comment:
this is in fact really good idea, to be able to enable the log thru the env, without config change

Reply:
enable it with env variable "AUTO_LOG_LEVEL"

Reply:
change AUTO_LOG_LEVEL to OPENVINO_LOG_LEVEL